libgobuster/statehelpers.go

```diff
@@ -86,7 +88,7 @@ func ValidateState(
 	}
 
 	if s.Mode == "dir" {
-		if err := ValidateDirModeState(s, extensions, codes, proxy, errorList); err.ErrorOrNil() != nil {
+		if err := ValidateDirModeState(s, extensions, codes, negcodes, proxy, errorList); err.ErrorOrNil() != nil {
```

Comment:
Given they are mutually exclusive, can we just pass in 'codes' and then the state of the `-s` or `-S` (possibly passed as a bool somewhere) determines whether we treat them as 'positive' or 'negative' code? Feels cleaner.